frontend/app/utils/route-utils.ts

```diff
@@ -100,3 +102,18 @@ export function getRouteByPath(pathname: string, routes: I18nRoute[]): I18nPageR
 function normalizePath(pathname: string): string {
   return pathname.replace(/\/+$/, '');
 }
+
+/**
+ * Recursively searches for a path matching the given file within a structure of I18nRoutes.
+ *
+ * @param i18nRouteFile - The file name of the route to search for.
+ * @param routes - The array of I18nRoutes to search within.
+ * @param request - The HTTP request object containing the URL from which to extract the locale.
+ * @returns The path of the route that matches the given file name.
+ */
+export function getPathByFile(i18nRouteFile: I18nRouteFile, routes: I18nRoute[], request: Request): string {
+  const route = getRouteByFile(i18nRouteFile, routes);
+  const lang = getLanguage(request);
+  const path = route.paths[lang ?? 'en'];
+  return path;
+}
```

Comment:
As suggested [here](https://github.com/DTS-STN/future-sir/pull/115#discussion_r1927531484), I think a better function would be `appRedirect(i18nRouteFile, request)`